src/v2/view-builder/views/ov/EnrollChannelPollDescriptionView.js

```diff
@@ -151,24 +171,19 @@ export default View.extend({
   getTemplateData() {
     const contextualData = this.options.appState.get('currentAuthenticator').contextualData;
     let enrolledDeviceName = '';
-    let deviceMap = {};
+    let deviceMap = getDeviceMap(this.options.appState);
     let showAnotherDeviceLink = false;
     if (contextualData) {
       if (contextualData?.devicebootstrap && contextualData?.devicebootstrap.enrolledDevices) {
         const enrolledDevices = contextualData?.devicebootstrap.enrolledDevices;
         enrolledDeviceName = Array.isArray(enrolledDevices) && !_.isEmpty(enrolledDevices) ?
           enrolledDevices[0] : enrolledDevices;
       }
-      if (contextualData.samedevice && contextualData.samedevice?.setupOVUrl) {
-        deviceMap = contextualData.samedevice;
-      } else if (contextualData.devicebootstrap && contextualData.devicebootstrap?.setupOVUrl) {
-        deviceMap = contextualData.devicebootstrap;
-      }
     }
 
     if (deviceMap.platform) {
       deviceMap.platformLC = deviceMap.platform.toLowerCase();
-      deviceMap.isDesktop = !(deviceMap.platformLC === 'ios' || deviceMap.platformLC === 'android');
+      deviceMap.isDesktop = getDeviceIsDesktop(deviceMap);
```

Comment:
Can all of the `deviceMap` mutation/setting logic happen inside `getDeviceMap`? I'd expect the returned value of that function to be a "complete" deviceMap object